docs/source/examples/best-practices.rst

```diff
@@ -0,0 +1,112 @@
+Best Practices
+==============
+
+
+Multi-GPUs
+~~~~~~~~~~
+
+When using multiple GPUs, if possible, it's best to co-locate as many GPUs on the same physical node.  This could be a
+DGX, a cloud instance with `multi-gpu options <https://rapids.ai/cloud>`_ , a high-density GPU HPC instance, etc.  This is done for
+two reasons:
+
+- 1. Moving data between GPUs is costly and performance decreases when computation stops due to communication overheads, Host-to-Device/Device-to-Host transfers, etc
+- 2. Mutli-GPU instances often come with accelerated networking like `NVLink <https://www.nvidia.com/en-us/data-center/nvlink/>`_.  These accelerated
+networking paths usually have much higher throughput/bandwidth compared with traditional networking *and* don't force and H-to-D/D-to-H transfers.  See `
+Accelerated Networking`_ for more discussion
+
+.. code-block:: python
+
+    from dask_cuda import LocalCUDACluster
+
+    cluster = LocalCUDACluster(CUDA_VISIBLE_DEVICES="0,1",
+                               protocol="ucx")
+
+GPU Memory Management
+~~~~~~~~~~~~~~~~~~~~~
+
+When using Dask-CUDA, especially with RAPIDS, it's best to use `RMM <https://docs.rapids.ai/api/rmm/stable/>`_ (RAPIDS Memory Manager)
+to pre-allocate memory on the GPU.  Allocating memory, while fast, takes a small amount of times, however, one can easily make
+hundreds of thousand or even millions of allocations in trivial workflows.  With RMM, allocations are sub-sampled from a larger
+pool and greatly reduces the allocation time and thereby increases performance:
+
+.. code-block:: python
+
+    from dask_cuda import LocalCUDACluster
+
+    cluster = LocalCUDACluster(CUDA_VISIBLE_DEVICES="0,1",
+                               protocol="ucx",
+                               rmm_pool_size="30GB")
+
+
+We also recommend allocating most, though not all, of the GPU memory space.  We do this so that temporary allocations, like those made deep inside CUDA libraries (libcudf, cupy, xgboost, etc) have room and the job is not halted due to an Out-Of-Memory (``OOM``) error.
```

Comment:
This is also not totally correct. cuDF and CuPy will use RMM to allocate memory, even temporary allocations (for CuPy that's only the case if it's internal memory management was replaced by RMM), I don't about xgboost. The problem with allocating the entire size is we don't know for example how large the CUDA context will be beforehand.

Reply:
Note as well that if `rmm_pool_size=None` then RMM will pick an initial pool for size `total_device_memory / 2`

Reply:
That isn't true. In Dask `rmm_pool_size=None` means it's disabled by default. The behavior you're thinking is within `rmm.reinitialize`, where passing `pool_allocator=True` will cause RMM to pick a pool with initial size of half the GPU's memory.

Reply:
Ah, sorry, yes.